### PR TITLE
Docgen from compiler

### PR DESCRIPTION
**Pull request by aspeddro**
This PR depends:

- Some improvements in compiler docstrings https://github.com/rescript-lang/rescript-compiler/pull/6417

TODO:
- [x] Update tests

### CODE REVIEW

**Review by zth — COMMENTED** [review]
Can I authorize deploy now?

**Review by zth — APPROVED** [review]
Approving to see if I can authorize.

**Comment by fhammerschmidt** [comment]
What's the state of this?

**Comment by aspeddro** [comment]
I'm still working on the UI

**Comment by aspeddro** [comment]
Ready for review

**Comment by cknitt** [comment]
Great stuff! 🎉 👍

A few minor things I noticed while trying it out:

* The new API docs show up under "v8.2.0", replacing the previous contents there. Shouldn't we leave the previous contents and add the new stuff under "v11.0.0"?
* Some internal modules are now there that weren't shown before (`Js.Internal`, `Js.MapperRt`). These should probably be suppressed somehow.
* `NullUndefined` is now `Null_undefined`.
* It's `TypedArray2`, but `Typed_array`, and there are `Float64_array` *and* `Float64Array` etc.
* The `Dom` module exports too much stuff (both `htmlDocument` and `_htmlDocument` etc.).

**Comment by aspeddro** [comment]
> The new API docs show up under "v8.2.0", replacing the previous contents there. Shouldn't we leave the previous contents and add the new stuff under "v11.0.0"?

It's like this on the master, I don't know why

> Some internal modules are now there that weren't shown before (Js.Internal, Js.MapperRt). These should probably be suppressed somehow.

We can suppress these modules

> NullUndefined is now Null_undefined.

Yes, the most current state of this PR shows the module `Null_undefined`

> It's TypedArray2, but Typed_array, and there are Float64_array and Float64Array etc.

I didn't understand this point

> The Dom module exports too much stuff (both htmlDocument and _htmlDocument etc.).

I don't know what we should do about this, I think it's better to keep

**Comment by aspeddro** [comment]
- Tests updated https://github.com/rescript-association/rescript-lang.org/pull/764/commits/cec8c48c8c9a0900125379a75f05add315e5e900
- `Js.Internal` and `Js.MapperRt` suppressed https://github.com/rescript-association/rescript-lang.org/pull/764/commits/d80ca3be964cd7dc728b13ffe0b721c29cd7e8e8

**Comment by aspeddro** [comment]
Dropdown menu fixed.
image

File: src/layouts/ApiLayout.res [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // This is used for the version dropdown in the api layouts
-let allApiVersions = [("latest", "v8.2.0"), ("v8.0.0", "< v8.2.0")]
+let allApiVersions = [("latest", "v11.0.0"), ("v10.0.0", "v8.2 - v10.1"), ("v8.0.0", "v6.0 - v8.2")]
```
Comment (cknitt) [inline src/layouts/ApiLayout.res:2]: Just noticed that there is also a `v9.0.0`directory:
https://github.com/rescript-association/rescript-lang.org/tree/master/pages/docs/manual/v9.0.0

What about that one, should it also be added?

Comment (cknitt) [reply src/layouts/ApiLayout.res:2]: Hmm, I think it should, the contents of the api subfolder of v9.0.0 is not identical with either v8.0.0 or v10.0.0.

Adding v9.0.0 would make the menu under "API" identical to the one under "Docs":

Bildschirmfoto 2024-01-07 um 09 29 03

Comment (aspeddro) [reply src/layouts/ApiLayout.res]: Done 70d8c2f7

**Comment by cknitt** [comment]
Just noticed something else regarding the new layout with the types and values in a column on the right:

Bildschirmfoto 2024-01-07 um 10 12 06

If the screen width is insufficient, that column disappears altogether:

Bildschirmfoto 2024-01-07 um 10 12 26

This seems not ideal compared to the old solution:

Bildschirmfoto 2024-01-07 um 10 13 08

**Pull request merged**